lms/djangoapps/program_enrollments/tests/test_tasks.py

```diff
@@ -0,0 +1,93 @@
+"""
+Unit tests for program_course_enrollments tasks
+"""
+from datetime import timedelta
+from django.test import TestCase
+from django.utils import timezone
+from freezegun import freeze_time
+from testfixtures import LogCapture
+import pytest
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment, ProgramCourseEnrollment
+from lms.djangoapps.program_enrollments.tasks import expire_waiting_enrollments, log
+from lms.djangoapps.program_enrollments.api.v1.tests.factories import (
+    ProgramCourseEnrollmentFactory, ProgramEnrollmentFactory
+)
+from student.tests.factories import UserFactory
+
+
+class ExpireWaitingEnrollmentsTest(TestCase):
+    """ Test expire_waiting_enrollments task """
+    def _setup_enrollments(self, external_user_key, user, created_date):
+        with freeze_time(created_date):
+            program_enrollment = ProgramEnrollmentFactory(
+                user=user,
+                external_user_key=external_user_key,
+            )
+            ProgramCourseEnrollmentFactory(
```

Comment:
I recommend also creating a course enrollment that's modified at `timezone.now()`, and making sure it still gets cleared when the expired program enrollment gets cleared.